Creating blog entry for Reaction Extensibility (#2844)
microsoft/teams-sdk: teams.md/blog/2026-05-18-reaction-extensbility-preview/index.md

---
slug: reaction-extensibility-preview
title: "Agents Can Now Send Reactions to Messages in Microsoft Teams"
date: 2026-05-18
authors:
  - name: Ricky Castaneda
    title: Microsoft
    url: https://github.com/MSFTRickyCastaneda
    image_url: https://github.com/MSFTRickyCastaneda.png
tags: [teams-sdk, reactions, preview, announcement, agents]
description: Your Teams agents can now react to messages with emoji - acknowledge, confirm, celebrate, and listen without sending a single extra reply.
---

**Public Preview | May 2026**

Your Teams agents can now react to messages with emojis - acknowledge, confirm, celebrate, and listen without sending a single extra reply. Available today in public preview across .NET, TypeScript, and Python.

> 

Reviewer: An image would be great here

Author: learn desktop image added

<!-- truncate -->

> ## The Problem: Agents Are Too Noisy

Reviewer: ```suggestion
## The Problem: Messages Can Be Too Noisy
```

Every interaction your agent has today requires a full message. A user submits a request? Your bot sends "Got it." A task completes? Another message. Someone asks a yes/no question? Yet another reply cluttering the thread.

Meanwhile, humans communicate volumes with a single 👍.

> Reaction Extensibility bridges that gap. Your agent can now participate in the lightweight, non-verbal layer of conversation that makes Teams collaboration feel natural.

Reviewer: ```suggestion
Agent reactions bridge that gap. Your agent can now participate in the lightweight, non-verbal layer of conversation that makes Teams collaboration feel natural.
```

In the slug too, _reaction extensibility_ makes this sound more complicated than it is.

>Your agent can now participate in the lightweight, non-verbal layer of conversation that makes Teams collaboration feel natural.

Great phrasing.

## What You Can Do

| Capability | Description |
| --- | --- |
| Add reactions | React to any message your agent can see - 1:1, group chat, or channel |
| Remove reactions | Programmatically remove reactions as context changes |
| Receive reaction events | Get notified when users react to your agent's messages |
| Emoji library | Discover EmojiIDs via the new Teams Reactions Reference |

## Real-World Scenarios

### 1. Instant Acknowledgment

User submits a request - agent reacts with 👍 ```(like)``` immediately. No "I'm on it" message required. The user knows they've been heard.

### 2. Progress Signaling

Agent reacts with one emoji while processing, then swaps to another on completion. One message, two state updates - zero thread noise.

```csharp
// Signal processing started
// Add reaction to message
await client.Send(
    new MessageReactionActivity()

>         .AddReaction(new Reaction() { Type = "happyface" })

Reviewer: Would love to see these suggested flows in the docs as guidance.

Have we put much thought/research into the emojis to be used for these scenarios, particularly this one? I'm not sure if it will be clear to users in the general case if a bot is using emoji reactions as hard status indicators, or if they will know to expect that they may change. Off the top of my head, I wonder if it would be clearer to use, say, an hourglass followed by a checkmark.

Author: changed to hourglass (holdon) -> checkmark (2705_whiteheavycheckmark)

        .WithReplyToId(activity.Id));

// ... task completes ...

// Swap to celebration
// Remove previoulsy added reaction
await client.Send(
    new MessageReactionActivity()
        .RemoveReaction(new Reaction() { Type = "happyface" })
        .WithReplyToId(activity.Id));

// Add new reaction to replace removed reaction
await client.Send(
    new MessageReactionActivity()
        .AddReaction(new Reaction() { Type = "like" })
        .WithReplyToId(activity.Id));
```

### 3. Silent Celebrations

Agent monitors a channel for deployment events. Successful deployment? React with 🎉. The team gets signal without additional thread clutter.

## Adding a Reaction

### TypeScript

```typescript
// Add a reaction
app.on('mention', async ({ activity, send }) => {
  await send(new MessageReactionActivity({
    replyToId: activity.id,
    reactions: [like]

>   }));

Author: @corinagum / @heyitsaamir can either of you confirm the edit here for me please :)

        Console.WriteLine($"User removed reaction: {reaction.Type}");}
});
```

### Python
```python
# Listen for a reaction added to agent message
@app.on_message_reaction
async def handle_reaction(ctx: ActivityContext[MessageReactionActivity]):
    for reaction in ctx.activity.reactions_added or []:
        print(f"User added reaction: {reaction.type}")
# Listen for a reaction removed from agent message
    for reaction in ctx.activity.reactions_removed or []:
        print(f"User removed reaction: {reaction.type}")
```

## Reaction IDs (EmojiIDs)

Reactions use Teams EmojiID strings - not Unicode emoji characters. Here are some examples:

| EmojiID | Emoji | Diverse (skin tones) |
| --- | --- | --- |
| `like` | 👍 | No |
| `heart` | ❤️ | No |
| `happyface` | 😊 | No |
| `grinningfacewithsmilingeyes` | 😄 | No |
| `1f603_grinningfacewithbigeyes` | 😃 | No |
| `1f44b_wavinghand` | 👋 | Yes |
| `1f590_handwithfingerssplayed` | 🖐️ | Yes |
| `1f91a_raisedbackofhand` | 🤚 | Yes |

Skin-tone variants are supported via suffix modifiers:

- `1f44b_wavinghand-tone1`
- `1f44b_wavinghand-tone2`
- `1f44b_wavinghand-tone3`
- `1f44b_wavinghand-tone4`
- `1f44b_wavinghand-tone5`

> ## Teams Reactions Reference
> 

Reviewer: ```suggestion
```

> Alongside this preview, we're publishing the [**Teams Reactions Reference**](https://learn.microsoft.com/en-us/microsoftteams/platform/agents-in-teams/teams-reactions-reference) - the canonical developer resource for discovering valid EmojiIDs supported by Teams.

Reviewer: ```suggestion
Alongside this preview, we're publishing the [**Teams Reactions Reference**](https://learn.microsoft.com/microsoftteams/platform/agents-in-teams/teams-reactions-reference) - the canonical developer resource for discovering valid EmojiIDs supported by Teams.
```

The reference includes:

- Complete catalog of supported EmojiIDs organized by category (Smileys, Hand gestures, etc.)
- Skin-tone variations for diverse emojis

> - EmojiID string format documentation

Reviewer: I don't think it has this?

Author: removed line 235 about format documentation

> 👉 [**Explore the Teams Reactions Reference**](https://learn.microsoft.com/en-us/microsoftteams/platform/agents-in-teams/teams-reactions-reference)

Reviewer: ```suggestion
👉 [**Explore the Teams Reactions Reference**](https://learn.microsoft.com/microsoftteams/platform/agents-in-teams/teams-reactions-reference)
```
